### PR TITLE
Updated readme.md file

### PR DESCRIPTION
**Pull request by ghduck**
corrected some spelling, added tables, rewrote some parts and added links

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
+  * string literals (e.g. `"John Gold"`)
+  * boolean literals (`true` and `false`)
+  * subpathes (e.g. `.nestedProp.deeplyNestedProp`)
 
-JSPath allows to use in predicate expressions following types of operators:
-  * comparison operators
-  * string comparison operators
-  * logical operators
-  * arithmetic operators
+Furthermore are following types of operators valide inside an object predicate:
```
Comment (dfilatov) [inline readme.md]: Seems should be "Furthermore the following types of operators are valid inside an object predicate:"

File: readme.md [hunk 2]
```diff
@@ -95,31 +78,59 @@ Result will be:
...
+You can use the wildcard symbol (`*`) instead of exact name of property:
+
+location step |  description                        
+------------- | -------------
+`.*`          | locates all properties immediately descended from the context items
+`..*`         | locates all properties deeply descended from the context items
  
-If you need to locate property containing non-alphanumerical characters, you can use quoted notation:
-  * ."property with non-alphanumerical characters"
+If you need to locate property containing non-alphanumerical characters, you have to quote the it:
```
Comment (dfilatov) [inline readme.md]: "the it"?

File: readme.md [hunk 3]
```diff
@@ -46,32 +46,15 @@ JSPath has been tested in IE6+, Mozilla Firefox 3+, Chrome 5+, Safari 5+, Opera
...
-    <td>Object</td>
-    <td>substitutions (optional)</td>
-  </tr>
-</table>
+
+parameter     |   data type        | description
+------------- |   -------------    | -------------
+`path`        | string             | [path expression](#documentation)
+`json`        | any valid JSON     | input JSON document
+`substs`      | JavaScript Object  | [substitutions (*optional*)](#substitutions)
```
Comment (dfilatov) [inline readme.md]: "Javascript" is redundant.

File: readme.md [hunk 4]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
-    <td>&lt;=</td>
-    <td>Returns true if the left operand is less than or equal to the right operand</td>
-    <td>.books{.id &lt;= 1}</td>
-  </tr>
-</table>
-
-Comparison rules:
+operator      |  description                                  | example                 
+------------- | -------------                                 | -------------
+`==`          | returns is `true` if both operands are equal    | `.books{.id == "1"}`
```
Comment (dfilatov) [inline readme.md]: "is" redundant

File: readme.md [hunk 5]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
-    <td>%</td>
-    <td>modulus</td>
-  </tr>
-</table>
+operator      |  description              
+------------- | ------------- 
+`+`          | addition        
+`-`          | subtraction  	  
+`*`          | multiplication 	
+`/`          | multiplication
```
Comment (dfilatov) [inline readme.md]: division

File: readme.md [hunk 6]
```diff
@@ -376,18 +278,24 @@ JSPath.apply('.books{.price < 17}.title', doc);
 ```
 
 ###Positional predicates###
-Positional predicates allow you to filter items by their context position.
-Positional predicates are always embedded in square brackets.
+Positional predicates allow you to filter items by their context position. All positional predicates are parenthesized by square brackets (`[` and `]`).
+
+There are four typs of positional predicates in JSPath also known as slicing methods:
```
Comment (dfilatov) [inline readme.md]: "typs" -> "types"

File: readme.md [hunk 7]
```diff
@@ -376,18 +278,24 @@ JSPath.apply('.books{.price < 17}.title', doc);
...
+
+operator      |  description                                  | example                 
+------------- | -------------                                 | -------------
+`[postion]`     | returns item in context on index `postion` (first item is at index `0`)    | `[3]` returns fourth item in context
```
Comment (dfilatov) [inline readme.md]: "postion" -> "position"

**Comment by ghduck** [comment]
Thanks :) I updated the file …

File: readme.md [hunk 8]
```diff
@@ -95,31 +78,59 @@ Result will be:
...
 
-Your location path can be absolute or relative.
-If location path starts with the root (^) you are using an absolute location path &mdash; your location path begins from the root items.
+location step                                          |  description
+-------------                                          | -------------
+`(.property1 | .property2 | .propertyN)`               | locates `property1`, `property2`, `propertyN` immediately descended from context items
+`(.property1 | .property2.property2_1.property2_1_1)`  | locates `.property1`, `.property2.property2_1.property2_1_1` immediately descended from context items
+
+Location paths can be absolute or relative.
+If location path starts with the caret (`^`) you are using an absolute location path. This syntax is used to locate a property when another context is already used in the location path and_or the object predicates.
```
Comment (dfilatov) [inline readme.md]: "and_or" -> "and/or"

File: readme.md [hunk 9]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
-    <td>*=</td>
-    <td>Like the '*==' but case insensitive</td>
-    <td>.books{.title *= "javascript"}</td>
-  </tr>
-</table>
+If both operands are strings, there're also available additional comparison operators:
+
+operator      |  description                                  | example                 
+------------- | -------------                                 | -------------
+`==`          | returns is `true` if both strings are equal   | `.books{.title == "clean code"}`
```
Comment (dfilatov) [inline readme.md]: "is" redundant

File: readme.md [hunk 10]
```diff
@@ -376,18 +278,24 @@ JSPath.apply('.books{.price < 17}.title', doc);
...
+JSPath supports four types of positional predicates – also known as slicing methods:
+
+operator      |  description                                  | example                 
+------------- | -------------                                 | -------------
+`[position]`  | returns item in context on index `position` (first item is at index `0`)    | `[3]` returns fourth item in context
+`[start:]`    | returns range of items whose index in context is greater or equal to `start`   |  `[2:]` returns items whose index is greater or equal to `2`
+`[:end]`    | returns range of items whose index in context is smaller than `end`    | `[:5]` returns first five items in context
+`[strat:end]`         | returns range of items whose index in context is greater or equal to `strat` and smaller than `end` | `[2:5]` returns three items on the indices `2`, `3` and `4`
+
+The `postion`, `start` or `end` index may be a negative number, which means it counts from the end of the array instead of the beginning:
```
Comment (dfilatov) [inline readme.md]: "postion" -> "position"

File: readme.md [hunk 11]
```diff
@@ -376,18 +278,24 @@ JSPath.apply('.books{.price < 17}.title', doc);
...
+Positional predicates allow you to filter items by their context position. All positional predicates are parenthesized by square brackets (`[` and `]`).
+
+JSPath supports four types of positional predicates – also known as slicing methods:
+
+operator      |  description                                  | example                 
+------------- | -------------                                 | -------------
+`[position]`  | returns item in context on index `position` (first item is at index `0`)    | `[3]` returns fourth item in context
+`[start:]`    | returns range of items whose index in context is greater or equal to `start`   |  `[2:]` returns items whose index is greater or equal to `2`
+`[:end]`    | returns range of items whose index in context is smaller than `end`    | `[:5]` returns first five items in context
+`[strat:end]`         | returns range of items whose index in context is greater or equal to `strat` and smaller than `end` | `[2:5]` returns three items on the indices `2`, `3` and `4`
```
Comment (dfilatov) [inline readme.md]: "strat" -> "start"

File: readme.md [hunk 12]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
-  </tr>
-  <tr>
-    <td>&lt;=</td>
-    <td>Returns true if the left operand is less than or equal to the right operand</td>
-    <td>.books{.id &lt;= 1}</td>
-  </tr>
-</table>
-
-Comparison rules:
+operator      |  description                                  | example
```
Comment (ghduck) [inline readme.md]: How about an introduction to comparison operators in JSPath?!

File: readme.md [hunk 13]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
-    <td>.books{!.title}</td>
-  </tr>
-</table>
-
-Logical operators convert their operands to boolean values using next rules:
-  * if operand is an array (as you remember result of applying subpath is also array):
-    * if length of array greater than zero, result will be true
-    * else result will be false
-  * Casting with double NOT (!!) javascript operator to be used in any other cases.
+operator      |  description                                  | example
```
Comment (ghduck) [inline readme.md]: Intro to logical operators?!

File: readme.md [hunk 14]
```diff
@@ -167,202 +178,93 @@ JSPath.apply('.books..name', doc);
...
-  <tr>
-    <td>/</td>
-    <td>division</td>
-  </tr>
-  <tr>
-    <td>%</td>
-    <td>modulus</td>
-  </tr>
-</table>
+operator      |  description
```
Comment (ghduck) [inline readme.md]: Intro to arithmetic operators and some examples?!

**Comment by dfilatov** [comment]
@ghduck good job!
Please squash all commits into one and then I'll merge PR.

**Comment by ghduck** [comment]
@dfilatov Thanks :smiley: 
As far as I know when you merge a pull request, [*you* can squash all commits into a single commit …](https://help.github.com/articles/about-pull-request-merge-squashing/)

**Comment by dfilatov** [comment]
@ghduck Thanks!

**Pull request merged**